### PR TITLE
[CCP] bar chart component

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
Adds little bar components that show the percent of patients that survive, die from cancer, and die from other causes.  Clicking on a row will make it `active` which currently has no visual indication except for that the bar charts in the other rows will use the `active` row as a reference when calculating the percent difference.  The bar charts are a visual representation of data regarding patients' treatment and outcome when compared to other methods of treatment over time.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/mcode-pilot/Visualizations/BarChart.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            textStyle = {"color":"#d9534f"};
+
+        }else{
+            // styles for when survival increases (green)
+            mainStyle={"width":`${oldPercent}%`, "backgroundColor":"#9e9e9e"};
+            changeStyle = {"width":`${percentIncrease}%`, "backgroundColor":"#5cb85c"}
+            deadStyle = {"width":`${cancerDead}%`, "backgroundColor":"white"}
+            otherDeadStyle = {"width":`${otherDead}%`, "backgroundColor":"#d1d1d1"}
+
+            textStyle = {"color":"#5cb85c"}; // t-shirt
```
Comment (noranda) [inline src/mcode-pilot/Visualizations/BarChart.jsx]: T-shirt? o.O

File: src/mcode-pilot/Visualizations/BarChart.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                <div className="prog-border" style={deadStyle}></div>
+                <div className="prog-fill" style={otherDeadStyle}></div>
+            </div>
+        );
+    }
+}
+
+BarChart.propTypes = {
+    values: PropTypes.object.isRequired,
+    compareTo: PropTypes.object
```
Comment (noranda) [inline src/mcode-pilot/Visualizations/BarChart.jsx]: Add `active` prop here.

File: src/mcode-pilot/Visualizations/BarChart.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        background-color: #9e9e9e;
+    }
+
+    .prog-fill-top { 
+        background-color: #e1e1e1;
+        width: 100px;
+    }
+
+    .arrow-up {
```
Comment (noranda) [inline src/mcode-pilot/Visualizations/BarChart.scss]: While this works, I'd recommend using [FontAwesome](https://fontawesome.com/icons/caret-up?style=solid) instead, as it's used throughout the project and is simple to change if needed.

File: src/mcode-pilot/Visualizations/BarChart.jsx (hunk 1 above)
Comment (KeeyanGhoreshi) [reply src/mcode-pilot/Visualizations/BarChart.jsx]: textStyle

**Comment by KeeyanGhoreshi** [comment]
In regards to [NB2], there seems to be some confusion regarding this edge case.  Here's an example that illustrates the "issue":

 Let's say surgery is 90% effective and chemotherapy is 40% effective.  

screen shot 2019-01-15 at 2 27 34 pm

To show the 50% decrease when using surgery as the reference value, we fill the chemotherapy bar 90% of the way full to match surgery's success rate and then color the excess red, in this case 50%.  

screen shot 2019-01-15 at 2 27 50 pm

The edge case occurs when the light gray section, the "non-cancer related deaths", is too large, and filling the bar to 90% would push into that section.  For example, if chemotherapy is 40% effective, with 45% cancer related deaths and 15% other deaths, we push 5% into the other deaths section.

screen shot 2019-01-15 at 2 28 33 pm
_____
 screen shot 2019-01-15 at 2 28 44 pm

So the question is, do we steal 5% from "other deaths", which would imply those deaths are caused by the treatment method, or do we match the white section instead?

For example surgery might cure 90% of people, with 5% dying from other causes, but of the people that could have died from cancer, it cured ~94%.  So for chemotherapy, we could fill the bar 94% of the way for potential-cancer deaths.  In other words, of the 85%, comparison by reference is 94%, so we fill the bar ~80% and leave the 15% of other deaths the way it is, giving us 5% leftover for "deaths from cancer", which perfectly mimics the amount of "deaths from cancer" as surgery.  This is visually unintuitive since the bars will not line up anymore when the percent of other deaths differs between categories. 

This of course is all predicated on the assumption that the "other causes" section can even vary between methods of the same year.  From all the designs I've seen, this doesn't seem to really be the case, as they all look pretty much the same in each column.

File: src/mcode-pilot/Visualizations/BarChart.jsx (hunk 1 above)
Comment (noranda) [reply src/mcode-pilot/Visualizations/BarChart.jsx]: I still don't follow. Can you clarify?

File: src/mcode-pilot/Visualizations/BarChart.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            textStyle = {"color":"#5cb85c"}; // t-shirt
+        }
+        return (
+            <div className="bar-chart">
+                <div>
+                    <p className="bar-chart-text">{Math.round(survived)}%</p>
+                    {!this.props.active?
+                        <p className="bar-chart-text right-text" style={roundedPercent!==0?textStyle:null}>
+                        {/* makes the tiny arrow */}
+                        <FontAwesome className={"tiny-arrow fas "} name={roundedPercent!==0?(roundedPercent>0?"caret-up":"caret-down"):null} />
```
Comment (noranda) [inline src/mcode-pilot/Visualizations/BarChart.jsx]: Don't need curly braces for the className.

```
<FontAwesome className="tiny-arrow fas" name={roundedPercent !==0 ? (roundedPercent > 0 ? "caret-up" : "caret-down") : null } />
```

**Comment by noranda** [comment]
[NB1] Resolved.

[NB2] Waiting on feedback from Dan from goInvo, but don't think it should hold up this PR.

This branch needs to be rebased.

**Pull request merged**